jenkins/docker/docker-copy.jenkinsfile

```diff
@@ -0,0 +1,78 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    options {
+        timeout(time: 1, unit: 'HOURS')
+    }
+    agent none  
+    parameters {
+        string(
+            name: 'SOURCE_IMAGE',
+            description: 'Complete source image path Eg: opensearchproject/<IMAGE_NAME>:<IMAGE_TAG>',
```

Comment:
Noneed for `opensearchproject` in description I think

Reply:
I think I know what you want to do but `SOURCE_IMAGE` and `DEST_IMAGE` are pointing two different things here.

The former is about registry+image+tag, while latter is image+tag makes it very confusing.

We should just add a param for SOURCE_IMAGE_REGISTRY just like DEST_IMAGE_REGISTRY, and remove the so-called `prod` or not boolean as the registry already define whether it is boolean or not.

Reply:
I have just added as an example, so that a user can pass the complete image path including registry, repo, imagename and tag.

Reply:
@peterzhuamazon The reason I dont want to add ` SOURCE_IMAGE_REGISTRY` is, I dont want to restrict to a particular registry, the source image can be open it can be pulled from any registry
Example if we have to copy some infra image to our registries we can add the complete image path to `SOURCE_IMAGE` and `DEST_IMAGE_REGISTRY` are fixed to only dockerhub/ECR hence I have added `DEST_IMAGE_REGISTRY` with prod flag (default is staging)
This was if we have to copy image from different registry example `1001235258/ostest`, still we should be able to do this.

Reply:
I have pushed the change in my latest commit @peterzhuamazon 
Added Source and destination registries to
[opensearchproject](https://hub.docker.com/u/opensearchproject),
[opensearchstaging](https://hub.docker.com/u/opensearchstaging),
[public.ecr.aws/opensearchproject](https://hub.docker.com/u/opensearchproject),
[public.ecr.aws/opensearchstaging](https://hub.docker.com/u/opensearchstaging).